File: server/datasets/python_source-code/Multiple-B2/organic_6.py
# Two Sum (hash map) - organic submission 6
# Recursive formulation, still backed by a hash map accumulator.

def two_sum_recursive(nums, target, index=0, seen=None):
    if seen is None:
        seen = {}
    if index >= len(nums):
        return None
    complement = target - nums[index]
    if complement in seen:
        return [seen[complement], index]
    seen[nums[index]] = index
    return two_sum_recursive(nums, target, index + 1, seen)


if __name__ == "__main__":
    nums = [2, 5, 5, 11]
    target = 10
    print("Indices:", two_sum_recursive(nums, target))

File: server/datasets/java_source-code/Multiple-B2/organic_6.java
// Two Sum (hash map) - organic submission 6
// Recursive formulation, still backed by a hash map accumulator.

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public static int[] twoSumRecursive(int[] nums, int target, int index, Map<Integer, Integer> seen) {
        if (index >= nums.length) {
            return null;
        }
        int complement = target - nums[index];
        if (seen.containsKey(complement)) {
            return new int[]{seen.get(complement), index};
        }
        seen.put(nums[index], index);
        return twoSumRecursive(nums, target, index + 1, seen);
    }

    public static int[] twoSum(int[] nums, int target) {
        return twoSumRecursive(nums, target, 0, new HashMap<>());
    }

    public static void main(String[] args) {
        int[] nums = {2, 5, 5, 11};
        int target = 10;
        System.out.println("Indices: " + Arrays.toString(twoSum(nums, target)));
    }
}
